src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -256,7 +256,17 @@ def list_containers(client, resource_group_name, vault_name, container_type="Azu
     return _get_containers(client, container_type, status, resource_group_name, vault_name)
 
 
-def check_protection_enabled_for_vm(cmd, vm_id):
+def check_protection_enabled_for_vm(cmd, vm_id=None, vm=None, resource_group_name=None):
+    if vm_id is None:
+        if is_valid_resource_id(vm):
+            vm_id = vm
+        else:
+            if vm is None or resource_group_name is None:
+                raise RequiredArgumentMissingError(
+                    """
+                    --vm or --resource-group missing. Please provide the required arguments.
+                    """)
```

Comment:
The indented spaces will also be shown in console. You can just use double quotes to continue the line:
```suggestion
                raise RequiredArgumentMissingError("--vm or --resource-group missing. Please provide the required "
                                                   "arguments.")
```